### PR TITLE
Use terracotta message tracker instead of our own implementation (close #2041)

### LINKED ISSUE #2041: Adopt terracotta message tracker
With the latest versions of Terracotta, there has been a number of changes.

Ehcache should follow these changes, adopting new APIs and features.

This specific issue is about adopting the Terracotta message tracker to replace the Ehcache one.

### PR DESCRIPTION
**Pull request by henri-tremblay**

### CODE REVIEW

File: clustered/server/src/main/java/org/ehcache/clustered/server/internal/messages/PassiveReplicationMessage.java [hunk 1]
```diff
@@ -33,41 +33,26 @@ public void setId(long id) {
     throw new UnsupportedOperationException("This method is not supported on replication message");
   }
 
-  public static class ClientIDTrackerMessage extends PassiveReplicationMessage {
+  public static class ChainReplicationMessage extends PassiveReplicationMessage implements ConcurrentEntityMessage {
+
     private final UUID clientId;
```
Comment (henri-tremblay) [inline clustered/server/src/main/java/org/ehcache/clustered/server/internal/messages/PassiveReplicationMessage.java]: Yes. It's not needed anymore. But I will remove the obsolete IDs afterwards.

File: clustered/client/src/test/java/org/ehcache/clustered/client/replication/ActivePassiveClientIdTest.java [hunk 2]
```diff
@@ -96,83 +68,4 @@ public void tearDown() throws Exception {
     clusterControl.tearDown();
   }
 
-  @Test
```
Comment (henri-tremblay) [inline clustered/client/src/test/java/org/ehcache/clustered/client/replication/ActivePassiveClientIdTest.java]: Good catch. Might be able to reuse it to test the sync message. Let's see

File: clustered/server/src/main/java/org/ehcache/clustered/server/store/ClusterTierActiveEntity.java [hunk 3]
```diff
@@ -43,24 +43,26 @@
 import org.ehcache.clustered.server.ServerStoreCompatibility;
```
Comment (henri-tremblay) [inline clustered/server/src/main/java/org/ehcache/clustered/server/store/ClusterTierActiveEntity.java]: `disconnected` you mean?

**Comment by henri-tremblay** [comment]
test this please

File: clustered/server/src/main/java/org/ehcache/clustered/server/internal/messages/EhcacheMessageTrackerMessage.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Message sending messages that are tracked for duplication. If a passive becoming active receives
+ * a duplicate, it needs to discard it.
+ */
+public class EhcacheMessageTrackerMessage extends EhcacheSyncMessage {
+
+  private final Map<ClientSourceId, Map<Long, EhcacheEntityResponse>> trackedMessages;
```
Comment (chrisdennis) [inline clustered/server/src/main/java/org/ehcache/clustered/server/internal/messages/EhcacheMessageTrackerMessage.java]: If we keep the `ClientSourceId`s as longs inside the message and delay conversion to ClientSourceId until inside the invoke then we can use the context to reconstruct the `ClientSourceId` instances and avoid creating our own implementation.

Comment (henri-tremblay) [inline clustered/server/src/main/java/org/ehcache/clustered/server/internal/messages/EhcacheMessageTrackerMessage.java]: Nice. I wasn't aware of `makeClientSourceId`

**Comment by henri-tremblay** [comment]
test this please

**Pull request merged**